packages/core/echo/echo-db/src/core-db/core-database.ts

```diff
@@ -323,7 +324,50 @@ export class CoreDatabase {
   }
 
   private _query(filter?: FilterSource, options?: QueryOptions) {
-    return new Query(new CoreDatabaseQueryContext(this, this._queryService), Filter.from(filter));
+    return new Query(
+      new CoreDatabaseQueryContext(this, this._queryService),
+      Filter.from(filter, { ...options, spaceIds: [this.spaceId] }),
+    );
+  }
+
+  // TODO(dmaretskyi): Mongo syntax.
+  async update(data: { id: string } & { [key: string]: any }) {
```

Comment:
what about deep updates `nested.object.field`?

Reply:
TBD. I wanna roll out this as a POC for people to experiment. Let's discuss API ideas in the afternoon today. I'm thinking we will just use the mongo syntax.